provider/resource_override.rb

```diff
@@ -0,0 +1,44 @@
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'api/object'
+
+module Provider
+  # Override to an Api::Resource in api.yaml
+  class ResourceOverride < Api::Object
+    attr_reader :description
+
+    def validate
+      super
+
+      check_optional_property :description, String
+    end
+
+    # Apply this override to the given instance of Api::Resource
+    def apply(api_resource)
+      extend_string api_resource, :description, @description
```

Comment:
@nelsonjr 

I prefer listing each property here rather than blindly extending all strings. Less magic, easier to read and reason about what the code is doing.

For that reason, using the `{{my-field}}` marker is fine.